server/model/monitor.js

```diff
@@ -53,7 +53,7 @@ class Monitor extends BeanModel {
         };
 
         if (this.sendUrl) {
-            obj.url = this.url;
+            obj.url = this.publicUrl ?? this.url; // Use publicUrl if available, otherwise use url
```

Comment:
```suggestion
            obj.url = this.publicUrl ?? this.url;
```

Reply:
Indeed, I removed it. But I wonder if the `publicUrl` aspect is still oblivious.

Reply:
Maybe add `https://github.com/louislam/uptime-kuma/pull/5435` URL as comment (but one line above)